packages/client/src/node.js

```diff
@@ -30,26 +30,29 @@ export let internalStateKeys = [
   'onMarkForUpdate',
   'afterSearch',
   'forceReplaceResponse',
+  'expand',
+  'collapse',
 ]
 
 export let autoKey = (x) => F.compactJoin('-', [x.field, x.type]) || 'node'
 
-export let initNode = _.curry(
-  ({ extend, types, snapshot }, dedupe, parentPath, node) => {
-    runTypeFunction(types, 'init', node, extend)
-    let key = dedupe(
-      node.key ||
-        runTypeFunctionOrDefault(autoKey, types, 'autoKey', node, extend)
-    )
-    extend(node, {
-      ..._.omit(_.keys(node), defaults),
-      // For some reason, type defaults can end up observable in real world apps, so we `snapshot` instead of `_.deepClone`
-      ..._.omit(_.keys(node), snapshot(getTypeProp(types, 'defaults', node))),
-      key,
-      path: [...parentPath, key],
-    })
-  }
-)
+export let initNode = _.curry((props, dedupe, parentPath, node) => {
+  let { types, extend, snapshot } = props
+  let actionProps = _.omit(['types'], props)
```

Comment:
Do we need to omit types? It's probably safe to pass it along 🤔

Reply:
Addressed